features/s3Transfer/s3TransferManager.feature

```diff
@@ -0,0 +1,149 @@
+    When I do upload this file with name <filename> with the specified part size of <partsize>
+    Then the object with name <filename> should have a total of <partnum> parts and its size must be <filesize>
+    Examples:
+      | filename                | filesize | partsize | partnum |
+      | myfile-test-3-1.txt     | 10485760 | 5242880  | 2       |
+      | myfile-test-3-2.txt     | 24117248 | 5242880  | 5       |
+      | myfile-test-3-3.txt     | 24117248 | 8388608  | 3       |
+
+  Scenario Outline: Successfully do multipart object upload from streams
+    Given I want to upload a stream of size <filesize>
+    When I do upload this stream with name <filename> and the specified part size of <partsize>
+    Then the object with name <filename> should have a total of <partnum> parts and its size must be <filesize>
+    Examples:
+      | filename                | filesize | partsize | partnum |
+      | myfile-test-4-1.txt     | 10485760 | 5242880  | 2       |
+      | myfile-test-4-2.txt     | 24117248 | 5242880  | 5       |
+      | myfile-test-4-3.txt     | 24117248 | 8388608  | 3       |
+
+  Scenario Outline: Does single object upload with custom checksum
+    Given I have a file with name <filename> and its content is <content>
+    When I upload this file with name <filename> by providing a custom checksum algorithm <checksum_algorithm>
+    Then the checksum from the object with name <filename> should be equals to the calculation of the object content with the checksum algorithm <checksum_algorithm>
+    Examples:
+      | filename                | content                            | checksum_algorithm |
+      | myfile-test-5-1.txt     | This is a test file content #1     | crc32              |
+      | myfile-test-5-2.txt     | This is a test file content #2     | crc32c             |
+      | myfile-test-5-3.txt     | This is a test file content #3     | sha256             |
+      | myfile-test-5-4.txt     | This is a test file content #4     | sha1               |
+
+  Scenario Outline: Does single object download
+    Given I have an object in S3 with name <filename> and its content is <content>
+    When I do a download of the object with name <filename>
+    Then the object with name <filename> should have been downloaded and its content should be <content>
+    Examples:
+      | filename                | content                            |
+      | myfile-test-6-1.txt     | This is a test file content #1    |
+      | myfile-test-6-2.txt     | This is a test file content #2    |
+      | myfile-test-6-3.txt     | This is a test file content #3    |
+
+  Scenario Outline: Successfully does multipart object download
+    Given I have an object in S3 with name <filename> and its size is <filesize>
+    When I download the object with name <filename> by using the <download_type> multipart download type
+    Then the content size for the object with name <filename> should be <filesize>
```

Comment:
Mentioned this in the context file, but we need assertions on the file/path the content is written to

Reply:
It is already tested in the [test](https://github.com/aws/aws-sdk-php/pull/3079/files#diff-edb622607799b89f33ea4333292c275516e8f56ce316d7377f2665fa594c54d3R441) implementation.